drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -0,0 +1,920 @@
+	if (buf == NULL) {
+		udc_submit_event(dev, UDC_EVT_ERROR, -ENOBUFS);
+		return -ENOBUFS;
+	}
+
+	if (udc_mcux_handler_zlt(dev, ep, buf, mcux_len)) {
+		return 0;
+	}
+
+	buf = udc_buf_get(dev, ep);
+
+	key = irq_lock();
+	udc_ep_set_busy(dev, ep, false);
+	irq_unlock(key);
+
+	if (buf == NULL) {
+		udc_submit_event(dev, UDC_EVT_ERROR, -ENOBUFS);
+		return -ENOBUFS;
+	}
+	if (ep == USB_CONTROL_EP_IN) {
+		err = udc_mcux_handler_ctrl_in(dev, buf, mcux_buf, mcux_len);
+	} else {
+		err = udc_mcux_handler_non_ctrl_in(dev, ep, buf, mcux_buf, mcux_len);
+	}
+
+	return err;
+}
+
+/* NXP MCUX controller driver notify transfers/status through this interface */
+usb_status_t USB_DeviceNotificationTrigger(void *handle, void *msg)
+{
+	usb_device_callback_message_struct_t *mcux_msg = msg;
+	uint8_t ep;
+	usb_device_notification_t mcux_notify;
+	struct udc_mcux_data *priv;
+	const struct device *dev;
+	usb_status_t mcux_status = kStatus_USB_Success;
+	int err = 0;
+
+	if ((NULL == msg) || (NULL == handle)) {
+		return kStatus_USB_InvalidHandle;
+	}
+
+	mcux_notify = (usb_device_notification_t)mcux_msg->code;
+	priv = (struct udc_mcux_data *)(PRV_DATA_HANDLE(handle));
+	dev = priv->dev;
+
+	switch (mcux_notify) {
+	case kUSB_DeviceNotifyBusReset:
+		struct udc_ep_config *cfg;
+
+		usb_mcux_control(dev, kUSB_DeviceControlSetDefaultStatus, NULL);
+		cfg = udc_get_ep_cfg(dev, USB_CONTROL_EP_OUT);
+		if (cfg->stat.enabled) {
+			udc_ep_disable_internal(dev, USB_CONTROL_EP_OUT);
+		}
+		cfg = udc_get_ep_cfg(dev, USB_CONTROL_EP_IN);
+		if (cfg->stat.enabled) {
+			udc_ep_disable_internal(dev, USB_CONTROL_EP_IN);
+		}
+		if (udc_ep_enable_internal(dev, USB_CONTROL_EP_OUT,
+					USB_EP_TYPE_CONTROL,
+					USB_MCUX_EP0_SIZE, 0)) {
+			LOG_ERR("Failed to enable control endpoint");
+			return -EIO;
+		}
+
+		if (udc_ep_enable_internal(dev, USB_CONTROL_EP_IN,
+					USB_EP_TYPE_CONTROL,
+					USB_MCUX_EP0_SIZE, 0)) {
+			LOG_ERR("Failed to enable control endpoint");
+			return -EIO;
+		}
+		udc_submit_event(dev, UDC_EVT_RESET, 0);
+		break;
+#if (defined(USB_DEVICE_CONFIG_ERROR_HANDLING) && (USB_DEVICE_CONFIG_ERROR_HANDLING > 0U))
+	case kUSB_DeviceNotifyError:
+		udc_submit_event(dev, UDC_EVT_ERROR, -EIO);
+		break;
+#endif
+#if (defined(USB_DEVICE_CONFIG_LOW_POWER_MODE) && (USB_DEVICE_CONFIG_LOW_POWER_MODE > 0U))
+	case kUSB_DeviceNotifySuspend:
+		udc_set_suspended(dev, true);
+		udc_submit_event(dev, UDC_EVT_SUSPEND, 0);
+		break;
+	case kUSB_DeviceNotifyResume:
+		udc_set_suspended(dev, false);
+		udc_submit_event(dev, UDC_EVT_RESUME, 0);
+		break;
+#if (defined(USB_DEVICE_CONFIG_LPM_L1) && (USB_DEVICE_CONFIG_LPM_L1 > 0U))
+	case kUSB_DeviceNotifyLPMSleep:
+		break;
+#endif
+#endif
+#if (defined(USB_DEVICE_CONFIG_DETACH_ENABLE) && (USB_DEVICE_CONFIG_DETACH_ENABLE > 0U))
+	case kUSB_DeviceNotifyDetach:
+		udc_submit_event(dev, UDC_EVT_VBUS_REMOVED, 0);
+		break;
+	case kUSB_DeviceNotifyAttach:
+		udc_submit_event(dev, UDC_EVT_VBUS_READY, 0);
+		break;
+#endif
```

Comment:
I looked at the definition of usb_device_notification_t and there is no reason for all the `#if-#endif`s. Please clean it up.